mithril-common/src/cardano_block_scanner/chain_reader_block_streamer.rs

```diff
@@ -0,0 +1,429 @@
+            ChainBlockNextAction::RollBackward { slot_number: 100 },
+        ])));
+        let mut block_streamer = ChainReaderBlockStreamer::try_new(
+            chain_reader,
+            Some(ChainPoint::new(100, 10, "hash-123")),
+            1,
+            MAX_ROLL_FORWARDS_PER_POLL,
+            TestLogger::stdout(),
+        )
+        .await
+        .unwrap();
+
+        let scanned_blocks = block_streamer.poll_next().await.expect("poll_next failed");
+        assert_eq!(None, scanned_blocks);
+    }
+
+    #[tokio::test]
+    async fn test_parse_expected_rollbackward_when_on_different_point_and_no_previous_rollforward()
+    {
+        let chain_reader = Arc::new(Mutex::new(FakeChainReader::new(vec![
+            ChainBlockNextAction::RollBackward { slot_number: 100 },
+        ])));
+        let mut block_streamer = ChainReaderBlockStreamer::try_new(
+            chain_reader,
+            None,
+            1,
+            MAX_ROLL_FORWARDS_PER_POLL,
+            TestLogger::stdout(),
+        )
+        .await
+        .unwrap();
+
+        let scanned_blocks = block_streamer.poll_next().await.expect("poll_next failed");
+
+        assert_eq!(Some(ChainScannedBlocks::RollBackward(100)), scanned_blocks,);
+
+        let scanned_blocks = block_streamer.poll_next().await.expect("poll_next failed");
+        assert_eq!(None, scanned_blocks);
+    }
+
+    #[tokio::test]
+    async fn test_parse_expected_rollforward_when_rollbackward_on_different_point_and_have_previous_rollforwards(
+    ) {
+        let chain_reader = Arc::new(Mutex::new(FakeChainReader::new(vec![
+            ChainBlockNextAction::RollForward {
+                parsed_block: ScannedBlock::new("hash-8", 80, 8, 1, Vec::<&str>::new()),
+            },
+            ChainBlockNextAction::RollForward {
+                parsed_block: ScannedBlock::new("hash-9", 90, 9, 1, Vec::<&str>::new()),
+            },
+            ChainBlockNextAction::RollForward {
+                parsed_block: ScannedBlock::new("hash-10", 100, 10, 1, Vec::<&str>::new()),
+            },
+            ChainBlockNextAction::RollBackward { slot_number: 9 },
+        ])));
+        let mut block_streamer = ChainReaderBlockStreamer::try_new(
+            chain_reader,
+            None,
+            1000,
+            MAX_ROLL_FORWARDS_PER_POLL,
+            TestLogger::stdout(),
+        )
+        .await
+        .unwrap();
+
+        let scanned_blocks = block_streamer.poll_next().await.expect("poll_next failed");
+
+        assert_eq!(
+            Some(ChainScannedBlocks::RollForwards(vec![
+                ScannedBlock::new("hash-8", 80, 8, 1, Vec::<&str>::new()),
+                ScannedBlock::new("hash-9", 90, 9, 1, Vec::<&str>::new())
+            ])),
+            scanned_blocks,
+        );
+    }
+
+    #[tokio::test]
+    async fn test_parse_expected_rollforward_when_rollbackward_on_different_point_and_does_not_have_previous_rollforwards(
```

Comment:
Little error, the test name is not consistent with what it assert.
```suggestion
    async fn test_parse_expected_backward_when_rollbackward_on_different_point_and_does_not_have_previous_rollforwards(
```